portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,49 +14,87 @@
 
 package com.google.sps.servlets;
 
-import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
+import com.google.sps.data.Comment;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.List;
-import java.util.ArrayList;
 import com.google.gson.Gson;
 
 /** Servlet that handles comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  private List<Comment> comments = new ArrayList<>();
 
-  /** Outputs JSON based on all user comments */
+  /** Retrieves and outputs JSON based on all user comments */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json");
+    // Create Query instance for Comment entities and sort based on time recieved 
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    // Create Datastore instance to interact with the Datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Contains all entities of type Comment in the Datastore 
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      
+      // Retrieve stored values from datastore
+      String name = (String) entity.getProperty("name");
```

Comment:
Just curious why do we need a cast here? What's the type of entity.getProperty?

Reply:
I read that it was because the type of the value returned by entity.getProperty() is not necessarily the same as type that was set by entity.setProperty() so the cast helps us be safe